### DISCUSSION TITLE
SSR Oauth

### DISCUSSION THREAD

**Discussion by Jerasaurus** [pocketbase/pocketbase#2434] [Q&A]
Hi! Im trying to make a simple application that authenticates a user through google oauth. I'm loosly following what fireship is doing here https://www.youtube.com/watch?v=gUYBFDPZ5qk but I'm trying to keep everything serverside so I don't have to expose pocketbase.  Is what I'm trying to do even possible? I have the auth data saved to a cookie and I'm loading it form a cookie using a sveltekit hook but I don't know how to access it on the client. 

The sad code is here [here](https://github.com/Recontaminated/PBSVC/tree/main/src) I feel like there's way too much for what just basic auth.

Any help would be amazing. Thank you!!

**Comment by ganigeorgiev** [comment]
> I'm trying to keep everything serverside so I don't have to expose pocketbase. Is what I'm trying to do even possible?

Yes, but will require reimplementing/proxying some of the PocketBase calls. This is also true if you use the realtime subscriptions since you'll need WebSocket/SSE connection between the browser<->node and then from node to establish an internal SSE connection with PocketBase with the SDK.

---

> I have the auth data saved to a cookie and I'm loading it form a cookie using a sveltekit hook but I don't know how to access it on the client.

Why do you want access from the client if you don't want to expose PocketBase?
In general you can access the cookie from the browser if the cookie is exported with `httpOnly: false`, but I'm not sure if this is necessary in your case.
Writing the cookie to the response after the OAuth2 redirect should be enough, which in your case would be handled by the SvelteKit hook (_I'm just not sure whether calling `throw redirect(303, '/')` will be intercepted and will continue the `hook.server.ts` handling as usual, aka. will execute the code after `await resolve(event)`, or will retrigger a new hook_).
